app/javascript/controllers/theme_controller.js

```diff
@@ -0,0 +1,45 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+  static targets = ["lightButton", "darkButton", "autoButton"]
+
+  connect() {
+    this.#applyStoredTheme()
+  }
+
+  setLight() {
+    this.#setTheme("light")
+  }
+
+  setDark() {
+    this.#setTheme("dark")
+  }
+
+  setAuto() {
+    this.#setTheme("auto")
+  }
+
+  get #storedTheme() {
+    return localStorage.getItem("theme") || "auto"
+  }
+
+  #setTheme(theme) {
```

Comment:
Could define a setter like

```js
set #theme(theme) { ... }
```

And then use like `this.#theme = "light"`

Reply:
Ah that's a nice pattern. Added in https://github.com/basecamp/fizzy/pull/2023/commits/2a0cf4efe9f56a626cd5bdb5c37865b8fda1c5b8